sos/report/plugins/__init__.py

```diff
@@ -1453,28 +1462,79 @@ def add_copy_spec(self, copyspecs, sizelimit=None, maxage=None,
         if isinstance(tags, str):
             tags = [tags]
 
+        def get_filename_tag(fname):
+            """Generate a tag to add for a single file copyspec
+
+            This tag will be set to the filename, minus any extensions
+            except '.conf' which will be converted to '_conf'
+            """
+            fname = fname.replace('-', '_')
+            if fname.endswith('.conf'):
+                return fname.replace('.', '_')
+            return fname.split('.')[0]
+
         for copyspec in copyspecs:
             if not (copyspec and len(copyspec)):
                 return False
 
-            if self.use_sysroot():
-                copyspec = self.path_join(copyspec)
-
-            files = self._expand_copy_spec(copyspec)
+            if not container:
+                if self.use_sysroot():
+                    copyspec = self.path_join(copyspec)
+                files = self._expand_copy_spec(copyspec)
+                if len(files) == 0:
+                    continue
+            else:
+                files = [copyspec]
 
-            if len(files) == 0:
-                continue
+            _spec_tags = []
+            if len(files) == 1:
+                _spec_tags = [get_filename_tag(files[0].split('/')[-1])]
 
-            def get_filename_tag(fname):
-                """Generate a tag to add for a single file copyspec
+            _spec_tags.extend(tags)
+            _spec_tags = list(set(_spec_tags))
 
-                This tag will be set to the filename, minus any extensions
-                except '.conf' which will be converted to '_conf'
-                """
-                fname = fname.replace('-', '_')
-                if fname.endswith('.conf'):
-                    return fname.replace('.', '_')
-                return fname.split('.')[0]
+            if container:
+                if isinstance(container, str):
+                    container = [container]
+                for con in container:
+                    if not self.container_exists(con):
+                        continue
+                    _tail = False
```

Comment:
What happens if we aim to copy *more* than sizelimit is? Then we set `_tail` to True and tail zero size from any other "container over limit"? Should not we set `_tail = False` before the `for con` cycle and have here `is _tail: break`?

Reply:
Since copying from containers currently only supports explicit files - no globs or similar - each `copyspec` is treated as a separate collection up to `sizelimit`, and is distinct from other containers. So if the stat of the file comes back over `sizelimit`, we set `_tail = True` for that collection within the container where the file is larger, and collect the last X bytes of the file.

Reply:
Ah I understand. I was already one step ahead, trying to collect `/foo/bar*` in one glob that one has to explicitly enumerate to `/foo/bar1`, `/foo/bar2` and `/foo/bar3` and `a_s_c` will collect up to `sizelimit` of each file - that is subtle different behaviour due to the "manual glob expansion". But it makes sense..